basic_broadcaster.go

```diff
@@ -30,8 +30,10 @@ func (b *BasicBroadcaster) Broadcast(seqNo uint64, data []byte) error {
 		b.working = true
 	}
 
-	b.lastMsg = &StreamDataMsg{SeqNo: seqNo, Data: data}
-	b.q <- b.lastMsg
+	latest := &StreamDataMsg{SeqNo: seqNo, Data: data}
+	b.lastMsgs = append(b.lastMsgs, latest)
+	b.lastMsgs = b.lastMsgs[1:]
+	b.q <- latest
```

Comment:
Is this correct? I'm not exactly sure of the usage of `b.q`, but it looks like you're setting `latest` to the currenly received data, but you're removing the first item from the `b.lastMsgs` array without using it. Do you want to insert `latest` into `b.q` or the removed oldest element from `lastMsgs`?

Reply:
We want to insert `latest` in this case.  `b.lastMsgs` is just a cache we keep around so we can send them to a new subscriber who hasn't gotten any segments yet.  For all other subscribers, we send them the latest segment.